qutebrowser/browser/hints.py

```diff
@@ -986,3 +1003,112 @@ def on_mode_left(self, mode):
             # hinting.
             return
         self._cleanup()
+
+
+class WordHinter(object):
+
+    """Generator for word hints.
+
+    Class attributes:
+
+    Attributes:
+
+    """
+
+    TAG_EXTRACTORS = {
+        "alt": lambda elem: elem["alt"],
+        "name": lambda elem: elem["name"],
+        "title": lambda elem: elem["title"],
+        "src": lambda elem: elem["src"].split('/')[-1],
+        "href": lambda elem: elem["href"].split('/')[-1],
+        "text": str,
+    }
+
+    TAGS_FOR = collections.defaultdict(
+        list, {
+            "IMG": ["alt", "title", "src"],
+            "A": ["title", "href", "text"],
+            "INPUT": ["name"]
+        }
+    )
+
+    FIRST_ALPHABETIC = re.compile('[A-Za-z]{3,}')
+
+    def __init__(self):
+        # will be initialized on first use.
+        self.words = set()
+
+    def ensure_initialized(self):
+        """Generate the used words if yet uninialized."""
+        if not self.words:
+            dictionary = config.get("hints", "dictionary")
+            try:
+                with open(dictionary, encoding="UTF-8") as wordfile:
+                    alphabet = set(string.ascii_lowercase)
+                    hints = set()
+                    lines = (line.rstrip().lower() for line in wordfile)
+                    for word in lines:
+                        if set(word) - alphabet:
+                            # contains none-alphabetic chars
+                            continue
+                        if len(word) > 4:
+                            continue
+                        for i in range(len(word)):
+                            hints.discard(word[:i + 1])
+                        hints.add(word)
+                    self.words.update(hints)
+            except IOError:
+                error = "Word hints require a dictionary at {}."
+                raise WordHintingError(error.format(dictionary))
```

Comment:
Make sure to add `str(e)` in the error message here somewhere (and do `except IOError as e:`). This might also be e.g. a permission problem (or any other I/O error), and it'd be frustrating for the user to debug if they don't see what's wrong.